compiler/il/VectorOperations.enum

```diff
@@ -18,6 +18,70 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mAnyTrue, \
+   /* .name                    = */ "mAnyTrue", \
+   /* .properties1             = */ 0, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
+   /* .properties4             = */ 0, \
+   /* .dataType                = */ TR::Int32, \
+   /* .typeProperties          = */ ILTypeProp::Size_4 | ILTypeProp::Integer, \
+   /* .childProperties         = */ ONE_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    test if any mask element is true */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mAllTrue, \
+   /* .name                    = */ "mAllTrue", \
+   /* .properties1             = */ 0, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
+   /* .properties4             = */ 0, \
+   /* .dataType                = */ TR::Int32, \
+   /* .typeProperties          = */ ILTypeProp::Size_4 | ILTypeProp::Integer, \
+   /* .childProperties         = */ ONE_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    test if all mask elements are true */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mmAnyTrue, \
+   /* .name                    = */ "mmAnyTrue", \
+   /* .properties1             = */ 0, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
```

Comment:
I think we need a flag to indicate that this operation is on a mask, under a mask

Reply:
Yes, I agree, will add.

Reply:
Added `ILProp1::VectorMasked` property. But after some thought, not sure about "on mask" property. Usually we check the return type of the opcode vs the type of any of its children. Do you think we need that flag? @BradleyWood

Reply:
You're right, its not necessary. I leave it up to you